src/App.tsx

```diff
@@ -1,110 +1,62 @@
 import React from 'react';
+import { useState } from 'react';
+
 import './App.css';
 import { getNumbers } from './utils';
+import { Pagination } from './components/Pagination';
+
+const options = [3, 5, 10, 20];
 
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const items: string[] = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [itemsPerPage, setItemsPerPage] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  function handleItemsPerPageChange(
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) {
+    setItemsPerPage(Number(event.target.value));
+    setCurrentPage(1);
+  }
+
   return (
     <div className="container">
       <h1>Items with Pagination</h1>
 
       <p className="lead" data-cy="info">
-        Page 1 (items 1 - 5 of 42)
+        {`Page ${currentPage} (items ${(currentPage - 1) * itemsPerPage + 1} - ${Math.min(currentPage * itemsPerPage, items.length)} of ${items.length})`}
       </p>
 
       <div className="form-group row">
         <div className="col-3 col-sm-2 col-xl-1">
           <select
+            value={itemsPerPage}
+            defaultValue={5}
```

Comment:
The `defaultValue` prop is unnecessary here because the `value` prop is already controlling the `<select>` component. You can remove `defaultValue={5}` to avoid confusion.